clang-tools-extra/docs/ReleaseNotes.rst

```diff
@@ -104,6 +104,296 @@ New check aliases
 Changes in existing checks
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+- Improved :doc:`bugprone-assert-side-effect
+  <clang-tidy/checks/bugprone/assert-side-effect>` check by detecting side
+  effect from calling a method with non-const reference parameters.
```

Comment:
still incorrectly rebased code, that stuff shoudn't be here
just rebase your changes on top of main branch, and then do force push, or cleanup this fill manually

Reply:
got it. Changed accordingly.

Reply:
Did I do mistake again...? I am really sorry if I did. I am new here. I am learning online and doing it. Please correct me @PiotrZSL